gis_exercise_2.py

```diff
@@ -178,7 +178,19 @@
 # hyperlink @sort: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html
 
 # I know I could sort the dataframe by user and timestamp at the same time, like this:
-userid2 = kruger_park_meters.groupby(['userid', 'timestamp'])
+km = kruger_park_meters
+by_user_and_time = km.sort_values(by=['userid', 'timestamp'])
+groups = by_user_and_time.groupby('userid', sort=True)
+
+def calculate_distance(rows):
+    if len(rows.geometry)>1:
+        return LineString(list(rows.geometry)).length
```

Comment:
what exactly is 'rows' here? I get that it is the rows of 'groups, but I don't understand the mechanism. Where do you tell python to iterate over rows of groups (and not e.g. over the columns?)